extra_tests/snippets/set.py

```diff
@@ -191,20 +191,15 @@ def __hash__(self):
 with assert_raises(TypeError):
 	a ^= [1,2,3]
 
-a = set([1, 2, 3])
```

Comment:
Note on this:

Set behaves the same as dictionaries and detects mutations even when they are balanced, i.e:

```
d = {1: 2, 3: 4}
dit = iter(d)
d[5] = 6
del d[6]
next(dit)  # RuntimeError
```

CPython doesn't detect any of these (neither for sets nor for dicts) but openly states in [PEP 3106](https://www.python.org/dev/peps/pep-3106/#id5) that mutations while iterating is undefined behavior:

> As in Python 2.x, mutating a dict while iterating over it using an iterator has an undefined effect and will in most cases raise a `RuntimeError` exception. (This is similar to the guarantees made by the Java Collections Framework.) 

So I'm pretty sure that a `RuntimeError` is the correct approach here.

Reply:
The fact that we detect it while CPython doesn't does make me believe we're currently doing more book-keeping than them and possibly slowing things down. I'll try and look into that too.